proposals/xxxx-wave-matrix.md

```diff
@@ -0,0 +1,396 @@
+  void ScalarDivide(ElTy Value);
+  void ScalarAdd(ElTy Value);
+  void ScalarSubtract(ElTy Value);
+};
+} // namespace detail
+```
+
+Each of these operations performs the corresponding element-wise arithmetic
+operation on the accumulator and the provided scalar value, storing the result
+back into the accumulator.
+
+### WaveMatrixAccumulator
+
+```c++
+namespace detail {
+template <typename T>
+struct is_8bit_packed_int_type =
+    std::enable_if_t<(std::is_same<T, int8_t4_packed>::value ||
+                      std::is_same<T, uint8_t4_packed>::value),
+                     std::true_type>;
+
+template <typename T> struct is_8bit_packed_int_type = std::false_type;
+
+template <typename T>
+struct is_32bit_int_type = std::enable_if_t<(std::is_same<T, int32_t>::value ||
+                                             std::is_same<T, uint32_t>::value),
+                                            std::true_type>;
+
+template <typename T> struct is_32bit_int_type = std::false_type;
+} // namespace detail
+
+template <typename ElTy, int NRows, int NCols>
+class WaveMatrixAccumulator
+    : WaveMatrixAccumulatorBase<ElTy, NRows, NCols, detail::WaveMatrixBase> {
+  void Add(WaveMatrixLeftColAcc<ElTy, NRows, NCols> LeftMatrix);
+  void Add(WaveMatrixRightRowAcc<ElTy, NRows, NCols> RightMatrix);
+  void Add(WaveMatrixAccumulator<ElTy, NRows, NCols> Matrix);
+
+  void Multiply(WaveMatrixLeft<ElTy, NRows, NCols> LHS,
+                WaveMatrixRight<ElTy, NRows, NCols> RHS);
+  void MultiplyAccumulate(WaveMatrixLeft<ElTy, NRows, NCols> LHS,
+                          WaveMatrixRight<ElTy, NRows, NCols> RHS);
+
+  template <typename MatElTy1, typename MatElTy2>
+  std::enable_if_t<detail::is_32bit_int_type<ElTy> &&
+                       detail::is_8bit_packed_int_type<MatElTy1> &&
+                       detail::is_8bit_packed_int_type<MatElTy2>,
+                   void>::type
+  Multiply(WaveMatrixLeft<MatElTy1, NRows, NCols> LHS,
+           WaveMatrixRight<MatElTy2, NRows, NCols> RHS);
+
+  template <typename MatElTy1, typename MatElTy2>
+  std::enable_if_t<detail::is_32bit_int_type<ElTy> &&
+                       detail::is_8bit_packed_int_type<MatElTy1> &&
+                       detail::is_8bit_packed_int_type<MatElTy2>,
+                   void>::type
+  MultiplyAccumulate(WaveMatrixLeft<MatElTy1, NRows, NCols> LHS,
+                     WaveMatrixRight<MatElTy2, NRows, NCols> RHS);
+};
+
+```
+
+#### Broadcast Add
+
+The `WaveMatrixAccumulator::Add` methods perform an element-wise addition of an
+accumulator matrix and the provided matrix or fragment accumulator.
+
+If the argument is a fragment object, the `Add` method broadcasts the argument
+accumulator up from an Mx1 or 1xN matrix to an MxN matrix then performs
+element-wise addition.
+
+#### WaveMatrixAccumulator::Multiply
+
+The `WaveMatrixAccumulator::Multiply` method performs multiplication of the left
+and right arguments and stores the result back into the `WaveMatrixAccumulator`.

```

Comment:
I know you specify that it "stores" the result, but given that this is the only one that doesn't accumulate (right?) maybe we should call out explicitly that it overwrites the contents of the accumulator